docs/oid-info.md

```diff
@@ -182,12 +182,12 @@ that Sigstore operates.
-| aud                | ??     | aud       | aud        | N/A                          | Only used to validate the JWT.                                                                                                                                                         |
+| aud                | aud     | aud       | aud        | N/A                          | Only used to validate the JWT.                                                                                                                                                         |
 | iss                | iss     | iss       | iss        | Issuer                       | This already exists. For example: https://token.actions.githubusercontent.com                                                                                                          |
 | exp                | exp     | exp       | exp        | N/A                          | Only used to validate the JWT.                                                                                                                                                         |
 | nbf                | nbf     | nbf       | nbf        | N/A                          | Only used to validate the JWT.                                                                                                                                                         |
 | iat                | iat     | iat       | iat        | N/A                          | Only used to validate the JWT.                                                                                                                                                         |
-| server_url + job_workflow_ref   | ??     | ??       | ??        | Build Signer URI             | Reference to specific build instructions that are responsible for signing. Can be the same as Build Config URI. For example a reusable workflow in GitHub Actions or a Circle CI Orbs. |
+| server_url + job_workflow_ref   | server_url + project_path + /-/blob/ + ci_config_sha + / + ci_config_path | ??       | ??        | Build Signer URI             | Reference to specific build instructions that are responsible for signing. Can be the same as Build Config URI. For example a reusable workflow in GitHub Actions or a Circle CI Orbs. |
 | job_workflow_sha   | ??     | ??       | ??        | Build Signer Digest          | An immutable reference to the specific version of the build instructions that is responsible for signing. Should include the digest type followed by the digest, e.g. `sha1:abc123`.   |
 | runner_environment | ??     | ??       | ??        | Runner Environment           | For platforms to specify whether the build took place in platform-hosted cloud infrastructure or customer-hosted infrastructure. For example: `platform-hosted` and `self-hosted`.     |
```

Comment:
```suggestion
| runner_environment |  runner_type  | ??       | ??        | Runner Environment           | For platforms to specify whether the build took place in platform-hosted cloud infrastructure or customer-hosted infrastructure. For example: `platform-hosted` and `self-hosted`.     |
```

Could be `instance`, `group`, or `project`.

Reply:
It would be useful to have runner ID and runner description in the claims as well. Can we put them here and concatenate it all together, or should they go in a different claim? e.g. `instance-11574084-3-green.shared-gitlab-org.runners-manager.gitlab.com`

Reply:
> My initial thought (assuming GitLab runner descriptions are predefined labels and not user generated content)

@wlynch runner description is an arbitrary string. We just happen to be using a string that looks like a FQDN for shared runners on gitlab.com.

---

@aladh @Brcrwilliams from a supply chain security perspective, you cannot trust the results of builds executed on self-hosted runners. A self-hosted runner may receive its build instructions, executes something entirely different, and sends us fake logs indicating otherwise.

This claim is used specifically in verifying that the build was executed on managed infrastructure where stronger guarantees can be made.

@wlynch any thoughts around self-hosted instances? We plan to support managed runners for GitLab Dedicated instances eventually and potentially offer managed runner fleets for self-hosted instances. We have no details on that yet, but I think it's safe to assume that "managed runner fleets" could be connected at any level (instance/group/project) and I would still expect them to be considered `platform-hosted`.

I think it's really the fleet itself that may be managed or self-hosted and that is independent of the instance it is connected to and at what level. Perhaps users should be establishing trust based on the hostname of the runner fleet/manager specifically instead of the abstract claim?

Reply:
Runner invocation may be a place to concatenate values together if there's other useful provenance claims

Note that the reason why we have runner_environment is so that clients can choose whether or not to trust the results - like you said, anything self-hosted should not be trusted for provenance metadata

Reply:
> @wlynch runner description is an arbitrary string. We just happen to be using a string that looks like a FQDN for shared runners on gitlab.com.

Can this value be arbitrarily modified by end users, or is it only set by the CI platform?

> @wlynch any thoughts around self-hosted instances?

+1 to what @haydentherapper said, though I think I have a slightly softer opinion in that trustworthy-ness of a runner pool is going to vary depending on the user. e.g.  A company may choose to trust it's own company-managed pool, but this may not be suitable for open source usage since it's opaque externally.
The important thing is to make sure end users have the data to distinguish these environments so they can write meaningful policy.

We generally have flexibility here, so long as the format is well-documented.

Reply:
> > @wlynch runner description is an arbitrary string. We just happen to be using a string that looks like a FQDN for shared runners on gitlab.com.
>
> Can this value be arbitrarily modified by end users, or is it only set by the CI platform?

This can be arbitrarily set by users via [the runner name](https://docs.gitlab.com/runner/configuration/advanced-configuration.html#the-runners-section). So, if we did it this way then it would be possible for self-hosted runners to impersonate platform-hosted runners.

> Perhaps users should be establishing trust based on the hostname of the runner fleet/manager specifically instead of an opinionated abstraction?

I don't think that a solution based on hostname is a good idea, because we don't have any domain validation mechanism and runners can be inside any network.

> +1 to what @haydentherapper said, though I think I have a slightly softer opinion in that trustworthy-ness of a runner pool is going to vary depending on the user. e.g. A company may choose to trust it's own company-managed pool, but this may not be suitable for open source usage since it's opaque externally.
The important thing is to make sure end users have the data to distinguish these environments so they can write meaningful policy.

In that case, I think that the values proposed by Ali have the most flexibility for this. We can provide: `runner_type`, `instance_url`, `group_id`, `project_id`. Policies can be defined for each runner type.

- `runner_type=instance`: validate based on `instance_url` (e.g. `https://gitlab.com`)
- `runner_type=group`: validate based on `instance_url` and `group_id`
- `runner_type=project`: validate based on `instance_url` and `project_id`

Reply:
@Brcrwilliams I was looking at this a bit and I am not sure there is value in including the `runner_type` claim at the moment. For `runner_type=group` does not tell you which group the runner is associated with. Likewise, `runner_type=project` runners can be shared across multiple projects: https://docs.gitlab.com/ee/ci/runners/runners_scope.html#enable-a-project-runner-for-a-different-project

I propose that we keep things simple for now and provide `runner_environment=gitlab-hosted|self-hosted` computed on our side.

Reply:
> I propose that we keep things simple for now and provide runner_environment=gitlab-hosted|self-hosted computed on our side.

Works for me. Let's do that.